vkcs/cdn/resource_resource/wait.go

```diff
@@ -0,0 +1,50 @@
+package resource_resource
+
+import (
+	"context"
+	"time"
+
+	"github.com/gophercloud/gophercloud"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/retry"
+	"github.com/vk-cs/terraform-provider-vkcs/vkcs/internal/services/cdn/v1/resources"
+)
+
+const ResourceReadyTimeout = 10 * time.Minute
+
+const resourceStatePollInterval = 10 * time.Second
+
+func WaitForResourceReady(ctx context.Context, client *gophercloud.ServiceClient, projectID string, resourceID int, timeout time.Duration) diag.Diagnostics {
+	var diags diag.Diagnostics
+
+	stateConf := &retry.StateChangeConf{
+		Pending:      []string{string(ResourceStatusProcessed)},
+		Target:       []string{string(ResourceStatusActive), string(ResourceStatusSuspended)},
```

Comment:
how does it works with letsencrypt? should we wait active status in this case?

Reply:
I think that status of the resource doesn't change with certificate being issued, but I added it from general considerations, like any other update operation. If you mean that the resource should be active to issue a Let's Encrypt cert, then we wait for it to be ready before applying any other changes on create/update. However, we do not wait for certificate to be issued, as this operation can last forever, but user can get info about its status using the corresponding `ssl_certificate.status` attribute.